WebCryptoAPI/generateKey/successes.js

```diff
@@ -65,7 +65,11 @@ function run_test(algorithmNames, slowTest) {
                     assert_goodCryptoKey(result, algorithm, extractable, usages, "secret");
                 }
             }, function(err) {
-                assert_unreached("Threw an unexpected error: " + err.toString());
+                if (err.toString().includes('not supported')) {
+                    test.skip(algorithm.name + ' not implemented');
```

Comment:
I am wondering if the better thing to do is to use the name of the error rather than assuming that it includes an english string is the correct behavior.  

if (err.name == "NotSupportedError")

Reply:
I too wondered this and discovered that `NotSupportedError` is not returned when the algorithm is not supported; it's an `OperationError`, which isn't distinct from the type used for genuine errors.

That's spec compliant, BTW: https://www.w3.org/TR/WebCryptoAPI/#SubtleCrypto-method-generateKey

Reply:
FWIW, I'm working on getting a less ugly/impractical use-case example from codec tests, in collaboration with @Orphis, so will remove this eye-sore once I have something nicer.

re: NotSupportedError, @ericroman920 informs me that 

> The conclusion was that Web Crypto implementations [must not throw NotImplemented](https://www.w3.org/Bugs/Public/show_bug.cgi?id=27598) (unless they don't support AES at all)